src/vmspawn/vmspawn.c

```diff
@@ -689,15 +690,15 @@ static int find_initrd(char **ret_initrd) {
         return -ENOENT;
 }
 
-static int run_virtual_machine(void) {
+static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
         _cleanup_(ovmf_config_freep) OvmfConfig *ovmf_config = NULL;
         _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
-
-        _cleanup_close_ int vsock_fd = -EBADF;
-        _cleanup_free_ char *machine = NULL, *qemu_binary = NULL, *mem = NULL, *trans_scope = NULL,
-                            *our_runtime_dir = NULL;
-
+        _cleanup_free_ char *qemu_binary = NULL, *mem = NULL, *trans_scope = NULL, *our_runtime_dir = NULL;
+        _cleanup_close_ int notify_sock_fd = -EBADF;
         _cleanup_strv_free_ char **cmdline = NULL;
+        _cleanup_free_ int *pass_fds = NULL;
```

Comment:
This variable doesn't seem to be actually passed to the qemu child process anywhere?

Reply:
The variable itself isn't but the FDs it stores are.
I use this variable to track which fds shouldn't be closed by safe_fork essentially.